File: CV_tasks/py/image_annotation.py
import cv2

def main():
    # Load an image from disk
    image = cv2.imread("images/test.webp")
    if image is None:
        print("Error: Couldn't load the image.")
        return

    text = "text"
    text_position = (50, 50)
    font_face = cv2.FONT_HERSHEY_SIMPLEX
    font_scale = 1.0
    text_color = (0, 255, 0)
    thickness = 2

    cv2.putText(image, text, text_position, font_face, font_scale, text_color, thickness)

    line_start = (100, 100)
    line_end = (300, 100)
    line_color = (0, 0, 255) 

    cv2.line(image, line_start, line_end, line_color, thickness)

    rect_top_left = (100, 150)
    rect_bottom_right = (300, 300)
    rect_color = (255, 0, 0)

    cv2.rectangle(image, rect_top_left, rect_bottom_right, rect_color, thickness)

    circle_center = (200, 400)
    radius = 50
    circle_color = (255, 255, 0) 

    cv2.circle(image, circle_center, radius, circle_color, thickness)

    cv2.imshow("Annotated Image", image)

    cv2.imwrite("annotated_image.jpg", image)

    cv2.waitKey(0)
    cv2.destroyAllWindows()

if __name__ == '__main__':
    main()

File: CV_tasks/Cpp/video-img-processing/image_annotation.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

int main() {
	cv::Mat img = cv::imread("../images/test.webp");

	if (img.empty()) {
		std::cerr << "Error: Could not open or read the image file." << std::endl;
		return 1;
	}

	std::string text = "text";
    cv::Point point(40, 40);
    int font = cv::FONT_HERSHEY_COMPLEX_SMALL;
    double font_size = 1.0;
    cv::Scalar color(45, 120, 165);
    int thickness = 2;
	cv::putText(img, text, point, font, font_size, color, thickness);

    cv::Point center(60, 40);
	int radius = 40;
	cv::circle(img, center, radius, color, thickness);

    cv::Point top_left_point(50, 100);
    cv::Point bottom_right_point(100, 200);
	cv::rectangle(img, top_left_point, bottom_right_point, color, thickness);

	cv::line(image, line_start, line_end, line_color, thickness);

    cv::Point rect_top_left(100, 150);
    cv::Point rect_bottom_right(300, 300);
    cv::Scalar rect_color(255, 0, 0);

	cv::imshow("Image", img);
	cv::waitKey(0);
	cv::destroyAllWindows();
}